src/forms/constraints/numbers.rs

```diff
@@ -0,0 +1,147 @@
+use crate::forms::FormValue;
+
+use super::FormConstraint;
+
+/// Validates that the input is an integer (positive or negative)
+pub struct Integer {
+    message: String,
+}
+
+impl Integer {
+    pub fn new(message: &str) -> Box<Self> {
+        Box::new(Self {
+            message: String::from(message),
+        })
+    }
+}
+
+impl FormConstraint for Integer {
+    fn validate(&self, output: &FormValue) -> bool {
+        match output {
+            FormValue::Nothing => true,
+            FormValue::Boolean(_) => true,
+            FormValue::Index(_) => true,
+            FormValue::String(value) => {
+                if let Some(chr) = value.chars().next() {
+                    if !chr.is_digit(10) && chr != '-' && chr != '+' {
+                        return false;
+                    }
+                }
+                value.chars().skip(1).all(|x| x.is_digit(10))
```

Comment:
You could do `value.parse::<i32>().is_ok()` instead. But if you want to keep it like that it's also fine.